### ISSUE TITLE
Problem generating application/octet-stream response

### ISSUE THREAD

**Issue by kivan-mih** [dart-ogurets/dart-openapi-maven#35]
When I have response in some endpoint in OpenAPI spec like this:
```
"responses" : {
  "200" : {
    "description" : "",
    "content" : {
      "application/octet-stream" : {
        "schema" : {
          "type" : "string",
          "format" : "binary"
        }
      }
    }
  }
}
```
dart generates code likes this:
```
return LocalApiClient.deserializeFromString(await utf8.decodeStream(response.body), 'MultipartFile') as MultipartFile;
```
which tries to deserialize bytes from string using utf-8 encoding. It's wrong since openapi spec implies that in this case we should receive plain bytes as the output of the api client. The code predictably throws FormatException when trying to invoke endpoint, since server sends raw bytes and generated client tries to deserialize utf-8 string from them

**Comment by rvowles** [comment]
Hey Kivan, this is inherited from the original Dart OpenAPI generator. Its going to take me a little while to sort this out for you as I am currently pretty busy with a FeatureHub release.

**Comment by kivan-mih** [comment]
Thx, binary data is important for me, since I plan to transfer stuff like images, different kinds of content bundles etc. Serializing them to base64 and return as string is not very good since it is guaranteed to be 4/3 larger than source bytes.

**Comment by rvowles** [comment]
Although it appears you are right @kivan-mih - the method does call the delegate which you can use directly and get an ApiResponse which stores your Stream<List<int>> which you need. That would get you around it while I fix this. I will change it so it always returns this ApiResponse directly.

**Comment by rvowles** [comment]
Leaving this ticket open until next release.

**Comment by rvowles** [comment]
fixed in 5.3
